ACLWeb.js

```diff
@@ -1,184 +1,196 @@
-		var bib = dirInUrl + fileRelPath + ".bib";
-		var pdf = dirInUrl + fileRelPath + ".pdf";
-		var j = fileRelPath.lastIndexOf("-");
-		var yearShort = fileRelPath.substring(j-2, j);
-		var year = "";
-		if (new Number(yearShort) < 50)
-			year = "20" + yearShort;
-		else
-			year = "19" + yearShort;
-
-		var attachments = new Array();
-		attachments.push({title:authorsShort + " " + year + ".pdf", mimeType:"application/pdf", url:pdf});
-
-		var type = "";
-		if (pageurl.indexOf("/anthology-new/J/")>-1)
-			type = "journalArticle";
-		else
-			type = "conferencePaper";
-
-		if (doImport)
-			callTranslator(bib, type, attachments);
-
+function extractFullProceedings(doc) {
+	let unwantedTitles = ['Front Matter', 'Author Index', 'Keyword Index'].map(function(title) { 
+		return 'not(contains(., "' + title + '"))';
+	}).join(' and ');
+	
+	let baseXpath = '//div[@id="content"]/p[i[' + unwantedTitles + ']]/'
+	
+	let ids = ZU.xpath(doc, baseXpath + 'a[@href = concat(text(), ".pdf")]');
+	ids = ids.map(function(id) { return id.textContent });
+	
+	let authors = ZU.xpath(doc, baseXpath + 'b');
+	authors = authors.map(function(author) { return author.textContent });
+	
+	let titles = ZU.xpath(doc, baseXpath + 'i');
+	titles = titles.map(function(title) { return title.textContent });
+	
+	let items = {};
+
+	for (let i = 0; i < ids.length; i++) {
+		let articleAuthors = authors[i].split('; ');
+		let authorSurname = articleAuthors[0].split(' ').pop();
+		let etAl = '';
+		if (articleAuthors.length > 1) {
+			etAl = ' et al.';
+		}
+		let author = authorSurname + etAl;
+		items[ids[i]] = ids[i] + ' (' + author + '): ' + titles[i];
 	}
+	
+	return items;
 }
 
-function callTranslator(bibFileURL, type, attachments) {
-	Zotero.Utilities.HTTP.doGet(bibFileURL, function(text) {
-
-		// load BibTex translator
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
-		translator.setString(text);
-		translator.setHandler("itemDone", function(obj, item) {
-			item.itemType = type;
-			item.attachments = attachments;
-			item.repository = "Association for Computational Linguistics"
-			item.complete();
+function scrapeBibtex(responseString, bibtexURL) {
+	let pdfURL = bibtexURL.replace('.bib', '.pdf');
+			
+	let translator = Zotero.loadTranslator("import");
+	translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
+	translator.setString(responseString);
+	translator.setHandler("itemDone", function (obj, item) {
+		item.attachments.push({
+			url: pdfURL,
+			title: 'Full Text PDF',
+			mimeType: 'application/pdf'
 		});
-		translator.translate();
+		delete item.itemID;
+		item.complete();
+	});
+	translator.translate();
+}
 
+function scrapeProceedings(doc, id) {
+	let newItem = new Zotero.Item("conferencePaper");
+	
+	let paragraphXpath = '//p[a[text()="' + id + '"]]/';
+	
+	let pdfURL = ZU.xpath(doc, paragraphXpath + 'a')[0].href;
```

Comment:
I suggest to filter here on the href attribute instead of relying on the order, i.e.
```js
let pdfURL = ZU.xpathText(doc, paragraphXpath + 'a[contains(@href, "pdf")]/@href');
```